docs/usage/configuration/config_documentation.md

```diff
@@ -3053,8 +3053,13 @@ Options for each entry include:
    values are `client_secret_basic` (default), `client_secret_post` and
    `none`.
 
+* `pkce_method`: Whether to use proof key for code exchange when requesting
+  and exchanging the token. Valid values are: `auto` or `always`. Defaults to
+  `auto`, which uses PKCE if supported during metadata discovery. Set to `always`
```

Comment:
first thoughts: are we missing a 'never' option?

Reply:
I didn't really want to add that unless we have to, but it is easy enough if you'd like. (I think it would just be a slight tweak to the load metadata code.)

Reply:
We [talked a bit about this in #synapse-dev:matrix.org](https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$gnzYAyGpZkj8VWWBHlVeQWT00y-lTyjA2cLT2opUIVA?via=matrix.org&via=element.io&via=envs.net), I think the tl;dr is that there's no value in disabling this unless you have an implementation which:

1. Advertises support for the `S256` variant of PKCE.
2. Incorrectly implements that support.

This seems unlikely to me, but does match what @sandhose suggested initially: https://github.com/matrix-org/synapse/pull/14750#discussion_r1059065530

I think the change is small, which makes sense to have an escape hatch to avoid blocking folks from upgrading.

Reply:
@reivilibre Would you mind taking another quick look at this?